ci: use LLVM/clang-16 in native_asan job

fanquake opened the pull request:
Similar to #27298. Working for me on `x86_64` and solves the issue I currently see with TSAN on `aarch64` with master (68828288e5d35c17848ab3da8cd231d1b69651c0):
```bash
crc32c/src/crc32c_arm64.cc:101:26: runtime error: load of misaligned address 0xffff84400406 for type 'uint64_t' (aka 'unsigned long'), which requires 8 byte alignment
0xffff84400406: note: pointer points here
 b9 c5 22 00 01 01  1a 6c 65 76 65 6c 64 62  2e 42 79 74 65 77 69 73  65 43 6f 6d 70 61 72 61  74 6f
             ^ 
    #0 0xaaaaaddaf0b4 in crc32c::ExtendArm64(unsigned int, unsigned char const*, unsigned long) src/./src/crc32c/src/crc32c_arm64.cc:101:26
    #1 0xaaaaadd2c838 in leveldb::crc32c::Value(char const*, unsigned long) src/./leveldb/util/crc32c.h:20:60
    #2 0xaaaaadd2c838 in leveldb::log::Reader::ReadPhysicalRecord(leveldb::Slice*) src/./src/leveldb/db/log_reader.cc:246:29
    #3 0xaaaaadd2ba9c in leveldb::log::Reader::ReadRecord(leveldb::Slice*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) src/./src/leveldb/db/log_reader.cc:72:38
    #4 0xaaaaadd41710 in leveldb::VersionSet::Recover(bool*) src/./src/leveldb/db/version_set.cc:910:19
    #5 0xaaaaadcf9fec in leveldb::DBImpl::Recover(leveldb::VersionEdit*, bool*) src/./src/leveldb/db/db_impl.cc:320:18
    #6 0xaaaaadd12068 in leveldb::DB::Open(leveldb::Options const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, leveldb::DB**) src/./src/leveldb/db/db_impl.cc:1487:20
    #7 0xaaaaad314e80 in CDBWrapper::CDBWrapper(DBParams const&) src/./src/dbwrapper.cpp:156:30
    #8 0xaaaaace94880 in CBlockTreeDB::CBlockTreeDB(DBParams const&) src/./txdb.h:89:23
    #9 0xaaaaace94880 in std::_MakeUniq<CBlockTreeDB>::__single_object std::make_unique<CBlockTreeDB, DBParams>(DBParams&&) /usr/bin/../lib/gcc/aarch64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34
    #10 0xaaaaace94880 in ChainTestingSetup::ChainTestingSetup(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<char const*, std::allocator<char const*> > const&) src/./src/test/util/setup_common.cpp:188:51
    #11 0xaaaaace95da0 in TestingSetup::TestingSetup(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<char const*, std::allocator<char const*> > const&, bool, bool) src/./src/test/util/setup_common.cpp:243:7
    #12 0xaaaaace96730 in TestChain100Setup::TestChain100Setup(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<char const*, std::allocator<char const*> > const&, bool, bool) src/./src/test/util/setup_common.cpp:274:7
    #13 0xaaaaac1ddbc8 in blockfilter_index_tests::BuildChainTestingSetup::BuildChainTestingSetup() src/./src/test/blockfilter_index_tests.cpp:26:8
    #14 0xaaaaac1ddbc8 in blockfilter_index_tests::blockfilter_index_initial_sync::blockfilter_index_initial_sync() src/./src/test/blockfilter_index_tests.cpp:112:1
    #15 0xaaaaac1ddbc8 in blockfilter_index_tests::blockfilter_index_initial_sync_invoker() src/./src/test/blockfilter_index_tests.cpp:112:1
    #16 0xaaaaabf08f7c in boost::function0<void>::operator()() const /usr/include/boost/function/function_template.hpp:763:14
    #17 0xaaaaabf95468 in boost::detail::forward::operator()() /usr/include/boost/test/impl/execution_monitor.ipp:1388:32
    #18 0xaaaaabf95468 in boost::detail::function::function_obj_invoker0<boost::detail::forward, int>::invoke(boost::detail::function::function_buffer&) /usr/include/boost/function/function_template.hpp:137:18
    #19 0xaaaaabf8e12c in boost::function0<int>::operator()() const /usr/include/boost/function/function_template.hpp:763:14
    #20 0xaaaaabe7be14 in boost::execution_monitor::catch_signals(boost::function<int ()> const&) /usr/include/boost/test/impl/execution_monitor.ipp:903:16
    #21 0xaaaaabe7c1c0 in boost::execution_monitor::execute(boost::function<int ()> const&) /usr/include/boost/test/impl/execution_monitor.ipp:1301:16
    #22 0xaaaaabe6f47c in boost::execution_monitor::vexecute(boost::function<void ()> const&) /usr/include/boost/test/impl/execution_monitor.ipp:1397:5
    #23 0xaaaaabe75124 in boost::unit_test::unit_test_monitor_t::execute_and_translate(boost::function<void ()> const&, unsigned long) /usr/include/boost/test/impl/unit_test_monitor.ipp:49:9
    #24 0xaaaaabed19fc in boost::unit_test::framework::state::execute_test_tree(unsigned long, unsigned long, boost::unit_test::framework::state::random_generator_helper const*) /usr/include/boost/test/impl/framework.ipp:815:44
    #25 0xaaaaabed0f6c in boost::unit_test::framework::state::execute_test_tree(unsigned long, unsigned long, boost::unit_test::framework::state::random_generator_helper const*) /usr/include/boost/test/impl/framework.ipp:784:58
    #26 0xaaaaabed0f6c in boost::unit_test::framework::state::execute_test_tree(unsigned long, unsigned long, boost::unit_test::framework::state::random_generator_helper const*) /usr/include/boost/test/impl/framework.ipp:784:58
    #27 0xaaaaabe73878 in boost::unit_test::framework::run(unsigned long, bool) /usr/include/boost/test/impl/framework.ipp:1721:29
    #28 0xaaaaabe9d244 in boost::unit_test::unit_test_main(boost::unit_test::test_suite* (*)(int, char**), int, char**) /usr/include/boost/test/impl/unit_test_main.ipp:250:9
    #29 0xffff8f0773f8  (/lib/aarch64-linux-gnu/libc.so.6+0x273f8) (BuildId: f37f3aa07c797e333fd106472898d361f71798f5)
    #30 0xffff8f0774c8 in __libc_start_main (/lib/aarch64-linux-gnu/libc.so.6+0x274c8) (BuildId: f37f3aa07c797e333fd106472898d361f71798f5)
    #31 0xaaaaabda55ac in _start (/home/fedora/ci_scratch/ci/scratch/build/bitcoin-aarch64-unknown-linux-gnu/src/test/test_bitcoin+0x10e55ac) (BuildId: b7909adaefd9db6cd6a7c4d3d40207cf6bdaf4b3)

SUMMARY: UndefinedBehaviorSanitizer: misaligned-pointer-use crc32c/src/crc32c_arm64.cc:101:26 in
```

fanquake committed:
ci: use LLVM/clang-16 in native_asan job

Similar to #27298.

fanquake commented:
native_fuzz split into #27363, as that is not blocked up upstream image issues/decisions.

fanquake commented on .cirrus.yml:
```diff
@@ -261,7 +261,7 @@ task:
   compute_engine_instance:
     image_project: ubuntu-os-cloud
-    image: family/ubuntu-2204-lts # when upgrading, check if we can drop "ADD_UNTRUSTED_BPFCC_PPA"
+    image: family/ubuntu-2304 # when upgrading, check if we can drop "ADD_UNTRUSTED_BPFCC_PPA"
```
I wonder if we should split the tracepoints stuff into it's own CI, so it's not a blocker for other unrelated tests.

fanquake replied:
Yep, will leave this as-is for now, and hopefully we can undraft on April 20th.

fanquake commented:
Lunar should be released on the 20th, https://discourse.ubuntu.com/t/lunar-lobster-release-schedule/27284, and hopefully the Gioogle Cloud images will be available shortly after that.

fanquake commented:
Ubuntu 23.04 has been released: https://lists.ubuntu.com/archives/ubuntu-announce/2023-April/000289.html. So the Google Cloud images should be available soon.

fanquake commented on .cirrus.yml:
```diff
@@ -269,7 +269,7 @@ task:
   compute_engine_instance:
     image_project: ubuntu-os-cloud
-    image: family/ubuntu-2204-lts # when upgrading, check if we can drop "ADD_UNTRUSTED_BPFCC_PPA"
+    image: family/ubuntu-2304 # when upgrading, check if we can drop "ADD_UNTRUSTED_BPFCC_PPA"
```
Pushed

fanquake replied:
Looks like either the naming is incorrect, or there isn't actually support for non-LTS?

fanquake replied:
@fkorotkov can you offer any insight here? Should we be able to use non-LTS Ubuntu images?

fkorotkov replied:
You can check for available images in ubuntu project like this:

```console
gcloud compute images list --project ubuntu-os-cloud --no-standard-images

NAME                                         PROJECT          FAMILY                         DEPRECATED  STATUS
ubuntu-1804-bionic-arm64-v20230418           ubuntu-os-cloud  ubuntu-1804-lts-arm64                      READY
ubuntu-1804-bionic-v20230418                 ubuntu-os-cloud  ubuntu-1804-lts                            READY
ubuntu-2004-focal-arm64-v20230302            ubuntu-os-cloud  ubuntu-2004-lts-arm64                      READY
ubuntu-2004-focal-v20230302                  ubuntu-os-cloud  ubuntu-2004-lts                            READY
ubuntu-2204-jammy-arm64-v20230421            ubuntu-os-cloud  ubuntu-2204-lts-arm64                      READY
ubuntu-2204-jammy-v20230421                  ubuntu-os-cloud  ubuntu-2204-lts                            READY
ubuntu-2210-kinetic-amd64-v20230125          ubuntu-os-cloud  ubuntu-2210-amd64                          READY
ubuntu-2210-kinetic-arm64-v20230125          ubuntu-os-cloud  ubuntu-2210-arm64                          READY
ubuntu-2304-lunar-amd64-v20230420            ubuntu-os-cloud  ubuntu-2304-amd64                          READY
ubuntu-2304-lunar-arm64-v20230420            ubuntu-os-cloud  ubuntu-2304-arm64                          READY
ubuntu-minimal-1804-bionic-arm64-v20230329   ubuntu-os-cloud  ubuntu-minimal-1804-lts-arm64              READY
ubuntu-minimal-1804-bionic-v20230329         ubuntu-os-cloud  ubuntu-minimal-1804-lts                    READY
ubuntu-minimal-2004-focal-arm64-v20230303    ubuntu-os-cloud  ubuntu-minimal-2004-lts-arm64              READY
ubuntu-minimal-2004-focal-arm64-v20230420    ubuntu-os-cloud  ubuntu-minimal-2004-lts-arm64              READY
ubuntu-minimal-2004-focal-v20230303          ubuntu-os-cloud  ubuntu-minimal-2004-lts                    READY
ubuntu-minimal-2204-jammy-arm64-v20230420    ubuntu-os-cloud  ubuntu-minimal-2204-lts-arm64              READY
ubuntu-minimal-2204-jammy-v20230420          ubuntu-os-cloud  ubuntu-minimal-2204-lts                    READY
ubuntu-minimal-2210-kinetic-amd64-v20230417  ubuntu-os-cloud  ubuntu-minimal-2210-amd64                  READY
ubuntu-minimal-2210-kinetic-arm64-v20230417  ubuntu-os-cloud  ubuntu-minimal-2210-arm64                  READY
ubuntu-minimal-2304-lunar-amd64-v20230420    ubuntu-os-cloud  ubuntu-minimal-2304-amd64                  READY
ubuntu-minimal-2304-lunar-arm64-v20230420    ubuntu-os-cloud  ubuntu-minimal-2304-arm64                  READY
```

Seem you need to use `family/ubuntu-2210-amd64`.

fanquake replied:
@fkorotkov Thanks for the info. Have pushed to use `family/ubuntu-2304-amd64`.

fanquake committed:
ci: remove usage of untrusted bpfcc-tools

We've migrated this job to Ubuntu 23.04, which
ships with newer versions of the tools:
https://packages.ubuntu.com/lunar/bpfcc-tools.

fanquake commented:
```bash
�[0;34m node0 2023-04-21T16:34:04.866961Z [http] [httpserver.cpp:308] [ThreadHTTP] [http] Exited http event loop �[0m
�[0;34m node0 2023-04-21T16:34:04.867449Z [shutoff] [httpserver.cpp:505] [StopHTTPServer] [http] Stopped HTTP server �[0m
�[0;34m node0 2023-04-21T16:34:04.868612Z [msghand] [logging.h:263] [error] ERROR: ProcessNewBlock: AcceptBlock FAILED (AcceptBlock: Failed to find position to write new block to disk) �[0m
�[0;34m node0 2023-04-21T16:34:04.869269Z [msghand] [util/thread.cpp:22] [TraceThread] msghand thread exit �[0m
�[0;34m node0 2023-04-21T16:34:04.889834Z [net] [util/thread.cpp:22] [TraceThread] net thread exit �[0m
�[0;36m test  2023-04-21T16:34:04.894000Z TestFramework.p2p (WARNING): Connection lost to 127.0.0.1:17120 due to [Errno 104] Connection reset by peer �[0m
�[0;34m node0 2023-04-21T16:34:04.894088Z [shutoff] [net.cpp:1526] [DumpAddresses] [net] Flushed 0 addresses to peers.dat  3ms �[0m
�[0;34m node0 2023-04-21T16:34:04.894130Z [shutoff] [net.cpp:593] [CloseSocketDisconnect] [net] disconnecting peer=53 �[0m
�[0;34m node0 2023-04-21T16:34:04.894402Z [shutoff] [net_processing.cpp:1544] [FinalizeNode] [net] Cleared nodestate for peer=53 �[0m
�[0;36m test  2023-04-21T16:34:04.897000Z TestFramework (ERROR): Assertion failed �[0m
�[0;36m                                   Traceback (most recent call last):�[0m
�[0;36m                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main�[0m
�[0;36m                                       self.run_test()�[0m
�[0;36m                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_block.py", line 1292, in run_test�[0m
�[0;36m                                       self.send_blocks(blocks, True, timeout=2440)�[0m
�[0;36m                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/feature_block.py", line 1435, in send_blocks�[0m
�[0;36m                                       self.helper_peer.send_blocks_and_test(blocks, self.nodes[0], success=success, reject_reason=reject_reason, force_send=force_send, timeout=timeout, expect_disconnect=reconnect)�[0m
�[0;36m                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 731, in send_blocks_and_test�[0m
�[0;36m                                       self.wait_until(�[0m
�[0;36m                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 463, in wait_until�[0m
�[0;36m                                       wait_until_helper(test_function, timeout=timeout, lock=p2p_lock, timeout_factor=self.timeout_factor)�[0m
�[0;36m                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 267, in wait_until_helper�[0m
�[0;36m                                       if predicate():�[0m
�[0;36m                                          ^^^^^^^^^^^�[0m
�[0;36m                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 460, in test_function�[0m
�[0;36m                                       assert self.is_connected�[0m
�[0;36m                                   AssertionError�[0m
�[0;36m test  2023-04-21T16:34:04.900000Z TestFramework (DEBUG): Closing down network thread �[0m
�[0;34m node0 2023-04-21T16:34:04.901483Z [scheduler] [util/thread.cpp:22] [TraceThread] scheduler thread exit �[0m
�[0;34m node0 2023-04-21T16:34:04.903621Z [shutoff] [kernel/mempool_persist.cpp:168] [DumpMempool] Writing 0 unbroadcast transactions to disk. �[0m
�[0;34m node0 2023-04-21T16:34:04.905760Z [shutoff] [kernel/mempool_persist.cpp:181] [DumpMempool] Dumped mempool: 2.009e-05s to copy, 0.00218644s to dump �[0m
�[0;34m node0 2023-04-21T16:34:04.905796Z [shutoff] [policy/fees.cpp:1011] [FlushUnconfirmed] [estimatefee] Recorded 0 unconfirmed txs from mempool in 6.309e-06s �[0m
�[0;34m node0 2023-04-21T16:34:04.907594Z [shutoff] [logging/timer.h:58] [Log] [bench] FlushStateToDisk: write block and undo data to disk started �[0m
�[0;36m test  2023-04-21T16:34:04.950000Z TestFramework (INFO): Stopping nodes �[0m
�[0;36m test  2023-04-21T16:34:04.950000Z TestFramework.node0 (DEBUG): Stopping node �[0m

�[0;34m node0 stderr Error: Disk space is too low! �[0m

�[1mTEST                                                   | STATUS    | DURATION

�[0m�[0;31mfeature_block.py                                       | ✖ Failed  | 59 s
�[0m�[0;31m�[1m
ALL                                                    | ✖ Failed  | 59 s (accumulated) 
�[0m�[0mRuntime: 59 s
```

fanquake commented on ci/test/00_setup_env_native_asan.sh:
```diff
@@ -19,8 +19,8 @@ else
 fi
```
Added here.

0xB10C commented:
Can't really comment on the change of using clang-16. The second commit dropping the bpfcc-tools from the PPA looks fine. You could remove the reference to this PR from the description of 5fbda45d2b962a67221ea0c8c35e38f6eef06499.

fanquake commented:
> You could remove the reference to this PR

Yep, dropped.

fanquake commented on .cirrus.yml:
```diff
@@ -269,7 +269,7 @@ task:
   # Images can be found here: https://cloud.google.com/compute/docs/images/os-details
   compute_engine_instance:
```
Applied, and we're no-longer seeing disk issues.

dergoegge commented:
utACK f952e679cd9c642e2c1f484ad8d75510ce25324c
